resolver-dns/src/main/java/io/netty/resolver/dns/DnsServerAddressStreamProviders.java

```diff
@@ -67,6 +116,21 @@ private DnsServerAddressStreamProviders() {
      * configuration.
      */
     public static DnsServerAddressStreamProvider platformDefault() {
+        if (STREAM_PROVIDER_CONSTRUCTOR != null) {
+            try {
+                return STREAM_PROVIDER_CONSTRUCTOR.newInstance();
+            } catch (IllegalAccessException e) {
+                // ignore
+            } catch (InstantiationException e) {
+                // ignore
+            } catch (InvocationTargetException e) {
+                // ignore
+            }
+        }
+        return unixDefault();
+    }
+
+    public static DnsServerAddressStreamProvider unixDefault() {
         return DEFAULT_DNS_SERVER_ADDRESS_STREAM_PROVIDER;
```

Comment:
On macOS users may never hit `DEFAULT_DNS_SERVER_ADDRESS_STREAM_PROVIDER`. Therefore, we may defer initialization until necessary, if we move this constant to another class.

Reply:
yep let me create an inner class.